components/layouts/dashboard/dashboardDrawer/dashboardDrawerItem/DashboardDrawerItem.tsx

```diff
@@ -0,0 +1,67 @@
+import styles from './DashboardDrawerItem.module.css';
+import {
+    Collapse,
+    ListItemButton, ListItemIcon, ListItemText, Tooltip,
+} from "@mui/material";
+import React from "react";
+import {ExpandLess, ExpandMore} from "@mui/icons-material";
+import {PopupState} from "material-ui-popup-state/core";
+import {bindToggle} from "material-ui-popup-state/hooks";
+
+
+export interface IDashboardDrawerItem {
+    text: string;
+    icon: React.ReactElement;
+    drawerOpen: PopupState;
+    dropDownType: string;
+
+    href?: string;
+
+    dropDownPopupState?: PopupState;
+}
+
+const DashboardDrawerItem: React.FC<IDashboardDrawerItem> = ({text, icon, drawerOpen, href, dropDownType, dropDownPopupState}) => {
+
+
+
+    return (
+      <div className={styles.container}>
+          <Tooltip title={text} placement="right">
+          {dropDownType.toLowerCase() === "child" && dropDownPopupState?
+              // If it is a Drop-Down Child
+              <Collapse in={dropDownPopupState.isOpen} timeout="auto" unmountOnExit className={drawerOpen ? styles.collapsibleClosed : styles.collapsibleOpened}>
+          <ListItemButton {...(href && { href: href })} className={drawerOpen.isOpen ? styles.listButtonClosed : styles.listButtonOpened}>
+              <ListItemIcon>
+                  <>{icon}</>
+              </ListItemIcon>
+              {drawerOpen.isOpen ? <ListItemText primary={text} className={styles.listText}/> : <></>}
```

Comment:
A fun way to do these conditional components is like this
```suggestion
              {drawerOpen.isOpen && <ListItemText primary={text} className={styles.listText}/>}
```